scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java

```diff
@@ -118,7 +124,7 @@ public RecoveredSchedulerState recover(long loadJobPeriod) {
         return new RecoveredSchedulerState(pendingJobs, runningJobs, finishedJobs);
     }
 
-    private void recoverRunningTasksOrResetToPending(InternalJob job, List<InternalTask> tasks) {
+    private void recoverRunningTasksOrResetToPending(InternalJob job, List<InternalTask> tasks, RMProxy rmProxy) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)